pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -465,6 +490,21 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 	return nil
 }
 
+func hasFinalizer(klusterlet *operatorapiv1.Klusterlet, finalizer string) bool {
+	for _, f := range klusterlet.Finalizers {
+		if f == finalizer {
+			return true
+		}
+	}
+	return false
+}
+
+func (n *klusterletController) addFinalizer(ctx context.Context, k *operatorapiv1.Klusterlet, finalizer string) error {
+	k.Finalizers = append(k.Finalizers, finalizer)
```

Comment:
need to check the finalizer has existed in the k.Finalizers ?

Reply:
every caller has checked the finalizer is not present, otherwise, this func should not be invoked.